eclipsePlugin/src/de/tobject/findbugs/reporter/ThrottledProgressMonitor.java

```diff
@@ -0,0 +1,90 @@
+package de.tobject.findbugs.reporter;
+
+import java.util.function.LongSupplier;
+
+import javax.annotation.Nonnull;
+import javax.annotation.concurrent.NotThreadSafe;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+
+/**
+ * <p>SpotBugs Eclipse Plugin uses {@link IProgressMonitor#setTaskName(String)} to tell what it is working for.
+ * It also updates progress quite frequently, which costs much time to update progress bar.</p> 
+ * 
+ * <p>Both of them make performance slow, so this class introduces throttling for better performance.</p>
+ * 
+ * @author Kengo TODA
+ * @version 3.1
+ */
+@NotThreadSafe
+class ThrottledProgressMonitor implements IProgressMonitor {
+	/**
+	 * Span of throttling, in milliseconds.
+	 */
+	static final int THROTTLE_SPAN = 300;
+	private static final long NOT_TRIGGERED = Integer.MIN_VALUE;
+
+	private final IProgressMonitor delegate;
+	private final LongSupplier currentTimeMillis;
+	private long lastRenamed = NOT_TRIGGERED;
+	private long lastWorked = NOT_TRIGGERED;
+	private int accumulatedWork = 0;
+
+	ThrottledProgressMonitor(@Nonnull IProgressMonitor delegate, @Nonnull LongSupplier currentTimeMillis) {
+		assert delegate != null;
+		assert currentTimeMillis != null;
+
+		this.delegate = delegate;
+		this.currentTimeMillis = currentTimeMillis;
+	}
+
+	@Override
+	public void setTaskName(String name) {
+		long currentTime = currentTimeMillis.getAsLong();
+		if (currentTime - lastRenamed >= THROTTLE_SPAN) {
```

Comment:
I would make sure we have set this at least once at the beginning. Otherwise it will not set the task name at all.

So something like if(lastRenamed == Integer.MIN_VALUE || ...)